### ISSUE TITLE
Disable format query in 1x1 convolution, and require specify input formats

### ISSUE THREAD

**Issue by CaoZhongZ** [oneapi-src/oneDNN#802]
# Summary
As title.

# Problem statement
1x1 convolutions have decent performance no matter what input formats are. However the query interface return blocking format forced a round of reorders. User could avoid it by specify format, but that requires a little knowledge of implementation.

# Preferred solution
Accept nchw/nhwc and blocked format as it is. If queried with any format, generate exception.

**Comment by vpirogov** [comment]
Hi @CaoZhongZ 

I'm not  sure I understand what disabling the quiery would gain in comparison to the current state. Applications that want to stick to plain activation format can do this by passing the format explicitly. 

Also note, that oneDNN now has full support for `nhwc` activations and this fact is reflected in documentation. There are two usage models:
1. Use any` to quiery formats and reorder the data if necessary
2. Create primitives explicitly with `nhwc` activations
Are there any scenarios when option #2 does not work?

**Comment by CaoZhongZ** [comment]
@vpirogov We focus on the first model.

Our preliminary test and analysis suggested if we receive input as it is, 1x1 convolutions have decent performance. Reorders + conv in blocked format can't outperform GEMM path in 1x1 cases with NCHW inputs, not exactly the case but very much like 'Relu', 'BN'. So the query gives a wrong impression that we have to do block format, which we made the mistake in our glue layer.

The request could be categorized as quality of life. We hope other use cases don't suffer the same mistake we have made.
